src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -435,30 +437,43 @@ CHIP_ERROR AccessControlAttribute::Write(const ConcreteDataAttributePath & aPath
 
 CHIP_ERROR AccessControlAttribute::WriteAcl(AttributeValueDecoder & aDecoder)
 {
+    FabricIndex accessingFabricIndex = aDecoder.AccessingFabricIndex();
+
     DataModel::DecodableList<AccessControlEntryCodec> list;
     ReturnErrorOnFailure(aDecoder.Decode(list));
 
-    size_t oldCount;
+    size_t oldCount = 0;
+    size_t allCount;
     size_t newCount;
     size_t maxCount;
-    ReturnErrorOnFailure(GetAccessControl().GetEntryCount(oldCount));
+
+    AccessControl::EntryIterator it;
+    AccessControl::Entry entry;
+    ReturnErrorOnFailure(GetAccessControl().Entries(it, &accessingFabricIndex));
+    while (it.Next(entry) == CHIP_NO_ERROR)
+    {
+        oldCount++;
+    }
+
+    ReturnErrorOnFailure(GetAccessControl().GetEntryCount(allCount));
     ReturnErrorOnFailure(list.ComputeSize(&newCount));
     ReturnErrorOnFailure(GetAccessControl().GetMaxEntryCount(maxCount));
-    ReturnErrorCodeIf(newCount > maxCount, CHIP_ERROR_INVALID_LIST_LENGTH);
+    VerifyOrReturnError(allCount >= oldCount, CHIP_ERROR_INTERNAL);
+    VerifyOrReturnError(static_cast<size_t>(allCount - oldCount + newCount) > maxCount, CHIP_ERROR_INVALID_LIST_LENGTH);
```

Comment:
Thanks.
@mlepage-google will we have more tests against the ACL cluster?

Reply:
This could be a YAML test right now, no?  Just try to write a single wildcard admin ACL, which should not break anything if it succeeds, and should succeed....

Reply:
We have pretty good unit test coverage of the underlying access control system module, and almost no coverage of the cluster itself. Yes we will have testing of the cluster, issue #10253 tracks that.

Reply:
Writing ACLs should work in master, I do that locally. Obviously it should work with this PR also (that's the goal).

Terminology note: this isn't what Boris meant, but to be clear for any readers, "wildcard write" is not a thing we support (at least for v1).